### DISCUSSION TITLE
[Toggle-group] Component concerns

### DISCUSSION THREAD

**Discussion by GomezIvann** [dxc-technology/halstack-react#951] [Ideas] [answered]
A few concerns with this component:

- Should the navigation between buttons be done by the tab key or using arrows? I know that each member is a button and using the tab key may be the most accurate option, but I think the arrow navigation fits this component better. 
- Another thing to discuss is the multiple selection variant, which may also affect the previous topic. Should this component allow to select more than one option? What is the use case that covers this functionality? Shouldn't you use another component if you want to allow multiple selections?

The main issue I see with this component is its concept since I can't figure out what use cases we are trying to cover with it. Is it an Actions Group, a Button Group, a Toggle Button Group, or a Content Switcher...? I think we need to define this a bit more this component.

To allow the user to select more than one option (and been a few options, not more than 1 or 2) we may need another kind of component:

1. New Checkbox group.
2. New Toggle Button Group component and implement an active/inactive button (toggle button).

Adobe Spectrum approach:

1. [Spectrum Action Group](https://spectrum.adobe.com/page/action-group/), which I think is the concept that we are trying to achieve with out toggle group.
2. [Spectrum Button Group](https://spectrum.adobe.com/page/button-group/).

Other approaches:

- https://carbondesignsystem.com/components/content-switcher/usage/
- https://baseweb.design/components/button-group/#radio-mode
- https://pharos.jstor.org/components/toggle-button-group
- https://www.w3.org/TR/wai-aria-practices/examples/switch/switch-button.html
- https://www.alexbuenodesign.com/blog/the_good_the_bad_and_the_toogle
- https://solid.buzzfeed.com/button-groups.html

**Comment by lcordobapuerto** [comment, marked as answer]
In my opinion, the use cases currently covered by the toggle group can be effectively handled by other components in our design system, such as the radio group or the switch, depending on the context. Additionally, since selections made with the toggle group are not submitted by default in a form, we should rethink this component to add more value and clearly differentiate it from others that serve a similar purpose.

After analyzing multiple UIs from other products, I agree with Iván that this component currently sits somewhere between what we already have and something closer to Spectrum’s [Action Group](https://spectrum.adobe.com/page/action-group/).
We could address this by redefining the component, ensuring it adds meaningful value to the design system and provides users with a solution that better fits the emerging use cases we’ve been seeing more frequently.

**Comment by GomezIvann** [comment]
New **toggle group** component with all the feedback applied!!! https://developer.dxc.com/halstack/next/components/toggle-group/

![image](https://64.media.tumblr.com/68d2ff5e684048df6f348a0b5a942220/tumblr_pqy1yl2TVx1wjld4so6_540.gif)
